### ISSUE TITLE
[XLA:GPU] dynamic-slice copy path writes a constant-offset `dynamic-update-slice` at the unclamped offset

### ISSUE THREAD

**Issue by snarang181** [openxla/xla#49120]
## Summary

A `dynamic-update-slice` with constant, out-of-range start indices is lowered on GPU through the dynamic-slice copy path, whose recorded `byte_offset` is computed from the raw starts. `DynamicUpdateSlice` semantics clamp each start into `[0, dim − update_dim]`; the copy path skips that, so the update is written at the wrong offset. Hence, the target window keeps stale values and the tail lands in the next row. 

## Reproducer

```python
import os; os.environ.setdefault("XLA_PYTHON_CLIENT_PREALLOCATE", "false")
import numpy as np, jax, jax.numpy as jnp
from jax import lax

shape = (3, 127, 1, 33)
x = (np.arange(int(np.prod(shape)), dtype=np.int64).reshape(shape) * 7919 % 10007 - 5000).astype(np.int32)

def f(x):
    t = x[:, 94]                                                 # static slice -> (3, 1, 33)
    upd = lax.dynamic_slice(t, [0, 0, 0], (1, 1, 8)) * 3 - 1      # update derived from the operand
    return lax.dynamic_update_slice(t, upd, [1, 1, 31])         # start (1, 1, 31) clamps to (1, 0, 25)

t = x[:, 94]; ref = t.copy(); ref[1, 0, 25:33] = t[0, 0, 0:8] * 3 - 1
for dev in {d.platform: d for d in jax.devices() + jax.devices("cpu")}.values():
    with jax.default_device(dev):
        out = np.asarray(jax.jit(f)(jnp.asarray(x)))
    bad = np.argwhere(out != ref)
    print(f"{dev.platform}: " + ("ok" if bad.size == 0 else f"WRONG {bad.shape[0]} elements; positions {[b.tolist() for b in bad]}"))
```

```
gpu: WRONG 14 elements; positions [[1,0,25] … [1,0,32], [2,0,0] … [2,0,5]]
cpu: ok
```

## Analysis

In the dump after `dynamic-slice-annotator` the DUS carries
`"dynamic_slice_config":{"byte_offset":"256","byte_stride":"0"}`: 256 B = 64 elements = `1·33 + 31`, the linearization of the unclamped starts; the clamped starts `(1, 0, 25)` give 58. `dynamic_slice_fusion_v2_thunk.cc` then applies only `ClampSliceOffset` against the whole buffer (`[0, buffer_size − slice_size]`), which 256 B passes. The per-dimension clamp is applied nowhere on this path; the regular loop emitter (used for traced starts and on CPU) clamps correctly. Relevant code: `dynamic_slice_analysis.cc` (`AnalyzeDynamicSlice`, byte offset from constant starts), `dynamic_slice_annotator.cc`, `dynamic_slice_fusion_v2_thunk.cc`. The path landed with PR #44044 (June 2026).

**Comment by penumallas1261** [comment]
Internal reproducer submitted in cl/985794605.

**Comment by akuegel** [comment]
I already have a potential fix for this: https://github.com/openxla/xla/pull/49309

Still needs to be reviewed.
